paddle/optimizer/Tensor.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+/**
+ * @brief tensor used by optimizer
+ */
+
+#include <string.h>
+#include <memory>
+#include "paddle/math/MemoryHandle.h"
+#include "paddle/utils/Common.h"
+#include "paddle/utils/Logging.h"
+
+namespace paddle {
+namespace optimizer {
+
+template <class T>
+class TensorT {
+public:
+  TensorT(size_t size)
+      : TensorT(std::make_shared<CpuMemoryHandle>(size * sizeof(float)), size) {
+  }
+  TensorT(CpuMemHandlePtr handle, size_t size)
+      : height_(1),
+        width_(size),
+        data_(reinterpret_cast<T*>(handle->getBuf())) {}
```

Comment:
that's right, I made a mistake here, Tensor have to save the shared_ptr for extending its lifetime. 
fix it, init data_ with data_ptr_.